src/trace-writer.ts

```diff
@@ -82,7 +82,9 @@ export class TraceWriter extends common.Service {
         config);
 
     this.logger = logger;
-    this.config = config;
+    // Clone the config object
+    this.config = Object.assign({}, config);
```

Comment:
Believe me. You can say `this.config = {...config}` and TS will transpile it correctly.

Reply:
You're right, I stand corrected.